webapp/src/components/surveyPost/surveyPost.tsx

```diff
@@ -43,6 +43,12 @@ function SurveyPost({post}: CustomPostTypeComponentProps) {
         }
     }, [responses]);
 
+    useEffect(() => {
+        if (!surveyExpired) {
+            client.refreshSurveyPost(post.id);
+        }
+    }, [post.id, surveyExpired]);
```

Comment:
The post id should never change, and we only expect the survey to move from not expired to expired, so... I would say this can have an empty dependency list (only runs on mount). But 0/5.

Reply:
Done.